### PR TITLE
Restore and deprecate make-container-max-widths mixin

### LINKED ISSUE #31436: 4.5.1 Breaking change, removed Mixing make-container-max-widths
### Version 4.5.1 made a breaking change removing a mixing 

The mixin is called: `make-container-max-widths`

ERROR LOG:

>>ERROR in ./src/styles.scss (./node_modules/css-loader/dist/cjs.js??ref--13-1!./node_modules/postcss-loader/src??embedded!./node_modules/sass-loader/dist/cjs.js??ref--13-3!./src/styles.scss)
Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: no mixin named make-container-max-widths
        on line 13 of node_modules/@my-components/styles/grid/_grid-container.scss
        from line 1 of C:\~\src\styles.scss
  @include make-container-max-widths();
 >>  -----------^

### OS/environment:
Windows, Angular 9, SCSS/SASS, bootstrap, node-sass

How to reproduce:

`.your-container {
    @include make-container-max-widths();
}`

### stackblitz example
https://stackblitz.com/edit/angular-ivy-2t6deq?file=src%2Fapp%2Fapp.component.scss

### LINKED ISSUE #31440: Deprecate make-container-max-widths mixin
After #31438 merges, let's add a deprecation notice to this mixin for v4.5.3.

### PR DESCRIPTION
**Pull request by mdo**
We removed this to de-dupe our grid containers in our compiled CSS (see #30969) and I wasn't thinking when reviewing that it'd be a breaking change. We're sticking with not using the mixin, but adding the mixin back for others to use.

We may need to add a deprecation notice to it as well.

Fixes #31436, fixes #31440

### CODE REVIEW

**Comment by grumbaut** [comment]
This is currently a blocker on a rather urgent project. Any chance it will be merged soon? If not, I'll just require 4.5.0 exactly for now.

**Comment by mdo** [comment]
Hoping to do a v4.5.2 tomorrow to patch this. Doing some digging and testing other things out. Lock in at v4.5.0 exactly for now to keep yourself unblocked if you can.

**Comment by XhmikosR** [comment]
@mdo Do we need the mixin in main, though? We could only target v4-dev where this was a breaking change, right?

**Comment by mdo** [comment]
Yup, we can retarget this one.

**Comment by XhmikosR** [comment]
Let's add the deprecation warning too so that we are covered since the mixin is removed from v5. @twbs/css-review in case you can help us here to move faster.

File: scss/mixins/_grid.scss [hunk 1]
```diff
@@ -18,6 +18,16 @@
...
 }
 
+// For each breakpoint, define the maximum width of the container in a media query
+@mixin make-container-max-widths($max-widths: $container-max-widths, $breakpoints: $grid-breakpoints) {
+  @each $breakpoint, $container-max-width in $max-widths {
+    @include media-breakpoint-up($breakpoint, $breakpoints) {
+      max-width: $container-max-width;
+    }
+  }
+  @include deprecate("The `make-container-max-widths` mixin", "v4.5.2", "v5", $ignore-warning);
```
Comment (MartijnCuppens) [inline scss/mixins/_grid.scss:28]: `$ignore-warning` is undefined, thus can be removed:

```suggestion
  @include deprecate("The `make-container-max-widths` mixin", "v4.5.2", "v5");
```

Comment (XhmikosR) [reply scss/mixins/_grid.scss]: True, bad CP.

**Review by MartijnCuppens — APPROVED** [review]

**Comment by nh2** [comment]
Is there any documentation or an upgrade note what one should use instead of `make-container-max-widths`?

**Comment by rchulyada** [comment]
would be good if there are list of depreciated and its replacement. Really annoying docs :(

**Pull request merged**